mitiq/shadows/quantum_processing.py

```diff
@@ -82,44 +82,48 @@ def random_pauli_measurement(
     circuit: cirq.Circuit,
     n_total_measurements: int,
     executor: Callable[[cirq.Circuit], MeasurementResult],
-) -> Tuple[NDArray[Any], NDArray[Any]]:
+    qubits: Optional[List[cirq.Qid]] = None,
+) -> Tuple[List[str], List[str]]:
     r"""
     Given a circuit, perform random Pauli measurements on the circuit and
-    return outcomes. The outcomes are represented as a string where a
-    z-basis measurement outcome of :math:`|0\rangle` corresponds to 1, and
-    :math:`|1\rangle` corresponds to -1.
+    return outcomes. The outcomes are represented as a tuple of two lists of
+    strings.
 
     Args:
         circuit: Cirq circuit.
         n_total_measurements: number of snapshots.
         executor: A callable which runs a circuit and returns a single
             bitstring.
+        qubits: The qubits to measure. If None, all qubits in the circuit
 
     Warning:
         The ``executor`` must return a ``MeasurementResult``
         for a single shot (i.e. a single bitstring).
 
     Returns:
-        Tuple of two numpy arrays. The first array contains
-        measurement outcomes (-1, 1) while the second array contains the
-        indices for the sampled Pauli's ("X", "Y", "Z").
-        This implies that local Clifford rotations plus z-basis measurements
-        are effectively equivalent to random Pauli measurements.
-        Each row of the arrays corresponds to a distinct snapshot or sample,
-        while each column corresponds to measurement outcomes
-        and random Pauli measurement on a different qubit.
+        Tuple of two list of string of length equals to`n_total_meausrements`,
+        where each element in the list is a string, with the following format:
+        strs in the 1st list: Circuit qubits computational basis
+        e.g. :math:`"01..":=|0\rangle|1\rangle..`.
+        strs in the 2ed list: The local Pauli measurement performed on each
+        qubit. e.g."XY.." means perform local X-basis measurement on the
+        1st qubit, local Y-basis measurement the 2ed qubit in the circuit.
```

Comment:
Here's my suggestion. I didn't want to use the suggestion tool since I cannot properly format the code here. I'll leave that to you.

> Tuple containing two lists of strings, each of length equal to `n_total_measurements`. Strings in the first list are sequences of 0's and 1's which represent qubit measurements in the computational basis (e.g. "01001"). Strings in the second list are sequences of Pauli operators which were performed on the circuit (e.g. "XZZII").

Reply:
Added suggestion with slightly abbreviated wording.